scripts/Muon/GUI/Common/corrections_tab_widget/background_corrections_presenter.py

```diff
@@ -0,0 +1,136 @@
+        self.view.set_slot_for_group_combo_box_changed(self.handle_selected_group_changed)
+        self.view.set_slot_for_show_all_runs(self.handle_show_all_runs_ticked)
+        self.view.set_slot_for_start_x_changed(self.handle_start_x_changed)
+        self.view.set_slot_for_end_x_changed(self.handle_end_x_changed)
+
+    def initialize_model_options(self) -> None:
+        """Initialise the model with the default fitting options."""
+        self.model.set_background_correction_mode(self.view.background_correction_mode)
+        self.model.set_selected_function(self.view.selected_function)
+
+    def handle_instrument_changed(self) -> None:
+        """User changes the selected instrument."""
+        self.model.set_background_correction_mode(BACKGROUND_MODE_NONE)
+        self.model.set_selected_function(FLAT_BACKGROUND)
+        self.view.background_correction_mode = BACKGROUND_MODE_NONE
+        self.view.selected_function = FLAT_BACKGROUND
+
+    def handle_runs_loaded(self) -> None:
+        """Handles when new run numbers are loaded into the interface."""
+        self.model.populate_background_corrections_data()
+
+    def handle_groups_changed(self) -> None:
+        """Handles when the selected groups have changed in the grouping tab."""
+        self.view.populate_group_selector(self.model.group_names())
+
+    def handle_run_selector_changed(self) -> None:
+        """Handles when the run selector is changed."""
+        self._update_displayed_corrections_data()
+
+    def handle_mode_combo_box_changed(self) -> None:
+        """Handles when the background corrections mode is changed."""
+        self.model.set_background_correction_mode(self.view.background_correction_mode)
+        self.view.set_background_correction_options_visible(not self.model.is_background_mode_none())
+
+    def handle_select_function_combo_box_changed(self) -> None:
+        """Handles when the selected function is changed."""
+        self.model.set_selected_function(self.view.selected_function)
+
+    def handle_selected_group_changed(self) -> None:
+        """Handles when the selected group has changed."""
+        self.model.set_selected_group(self.view.selected_group)
+        self._update_displayed_corrections_data()
+
+    def handle_show_all_runs_ticked(self) -> None:
+        """Handles when the show all runs check box is ticked or unticked."""
+        self.model.set_show_all_runs(self.view.show_all_runs)
+        self._update_displayed_corrections_data()
+
+    def handle_start_x_changed(self) -> None:
+        """Handles when a Start X table cell is changed."""
+        run, group = self.view.selected_run_and_group()
+        self._check_start_x_is_valid_for(run, group)
+
+        self.model.set_start_x(run, group, self.view.start_x(run, group))
+        self.model.set_end_x(run, group, self.view.end_x(run, group))
+
+    def handle_end_x_changed(self) -> None:
```

Comment:
if we check both  the start and end x are valid, this and the above could be a single function